Fixes #57515 - Fix test_async_write_logs_should_execute_successfully test

hwang-cadent opened the pull request:
## Description

This PR fixes the broken test `test_async_write_logs_should_execute_successfully()` in the KubernetesPodOperator test suite. The test was incorrectly implemented and was not properly testing the async log writing functionality.

## Changes

* Added `@patch(KUB_OP_PATH.format("client"))` decorator to properly mock `self.client.read_namespaced_pod_log()` which is the actual method used by `_write_logs()`
* Fixed mock setup to use `mocked_client.read_namespaced_pod_log` instead of incorrectly mocking `mock_manager.read_pod_logs` (which is never called)
* Fixed mock return value to return an iterable of bytes: `[test_logs.encode("utf-8")]` as expected by the implementation
* Added `mock_manager.await_pod_completion.return_value` for proper test flow
* Fixed assertions to verify actual log output using `caplog` instead of a meaningless string assertion
* Fixed `get_logs=False` assertion to check `mocked_client.read_namespaced_pod_log.assert_not_called()` instead of the incorrect `pod_manager.read_pod_logs` check

## Root Cause

The test had three main issues:
1. **Wrong mock target**: It mocked `pod_manager.read_pod_logs` but `_write_logs()` actually calls `self.client.read_namespaced_pod_log()` directly
2. **Incorrect assertion**: The assertion `assert f"Container logs: {test_logs}"` always passes (it's just checking if a string exists, not if it was logged)
3. **Wrong verification for get_logs=False**: It checked `pod_manager.read_pod_logs` which is never called by `_write_logs()`

## Testing

* Both parameterized test cases now pass:
  - `test_async_write_logs_should_execute_successfully[True]` ✅
  - `test_async_write_logs_should_execute_successfully[False]` ✅
* All existing tests continue to pass

## Type of Change

* Bug fix (non-breaking change which fixes an issue)

Fixes #57515

hwang-cadent committed:
Fix test_async_write_logs_should_execute_successfully test

- Add client mock patch to properly mock self.client.read_namespaced_pod_log
- Fix assertions to verify actual log output using caplog
- Fix get_logs=False assertion to check client.read_namespaced_pod_log instead of pod_manager.read_pod_logs
- Add await_pod_completion mock return value

Fixes #57515

hwang-cadent committed:
Merge branch 'main' into fix-57515-kubernetes-pod-operator-test

jscheffl commented on providers/cncf/kubernetes/tests/unit/cncf/kubernetes/operators/test_pod.py:
```diff
@@ -2595,17 +2595,19 @@ def test_async_get_logs_should_execute_successfully(
     @patch(KUB_OP_PATH.format("extract_xcom"))
     @patch(HOOK_CLASS)
     @patch(KUB_OP_PATH.format("pod_manager"))
     @pytest.mark.xfail
```
If the test is fixed by the PR, can you remove the xfail flag? Then CI should turn green as a proof
```suggestion
```

hwang-cadent replied:
Thank you, @jscheffl. Removed the @pytest.mark.xfail decorator and fixed an incorrect assertion. Both test cases now pass. The test is working correctly.

hwang-cadent committed:
Remove @pytest.mark.xfail and fix incorrect assertion

- Remove @pytest.mark.xfail decorator as the test is now fixed
- Remove incorrect assertion checking mock_manager.read_pod_logs which
  is never called by _write_logs() (it calls client.read_namespaced_pod_log
  directly instead)

Both parameterized test cases now pass:
- test_async_write_logs_should_execute_successfully[True] ✅
- test_async_write_logs_should_execute_successfully[False] ✅

Addresses reviewer feedback to remove xfail flag as proof the test works.

hwang-cadent committed:
Merge branch 'main' into fix-57515-kubernetes-pod-operator-test

hwang-cadent committed:
Apply ruff formatting fixes

- Remove unused imports: BytesIO and HTTPResponse
- Format function parameters on multiple lines for better readability

hwang-cadent committed:
Merge branch 'main' into fix-57515-kubernetes-pod-operator-test

hwang-cadent committed:
Merge branch 'main' into fix-57515-kubernetes-pod-operator-test

hwang-cadent committed:
Merge branch 'main' into fix-57515-kubernetes-pod-operator-test

hwang-cadent committed:
Merge branch 'main' into fix-57515-kubernetes-pod-operator-test

jscheffl reviewed (approved):
Looks good! LGTM when CI turns green!

hwang-cadent committed:
Merge branch 'main' into fix-57515-kubernetes-pod-operator-test

hwang-cadent committed:
Merge branch 'main' into fix-57515-kubernetes-pod-operator-test
